### PR TITLE
Code to fix the Optional Null

### PR DESCRIPTION
**Pull request by sarangan12**
This is to fix the issue reported in https://github.com/Azure/autorest.typescript/issues/909.

@joheredi Please review and approve. 

@ramya-rao-a FYI....

### CODE REVIEW

File: test/integration/generated/bodyComplex/src/models/index.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export type SharkUnion = Shark | Sawshark | Goblinshark | Cookiecuttershark;
 
 export interface BasicDef {
   /** Basic Id */
-  id?: number | null;
```
Comment (sarangan12) [inline test/integration/generated/bodyComplex/src/models/index.ts:19]: Per Autorest, this is a arm swagger. So this change is expected.

File: test/integration/generated/bodyComplexWithTracing/src/models/index.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export type SharkUnion = Shark | Sawshark | Goblinshark | Cookiecuttershark;
```
Comment (sarangan12) [inline test/integration/generated/bodyComplexWithTracing/src/models/index.ts:16]: Per Autorest, this is a arm swagger. So this change is expected.

File: src/typescriptGenerator.ts [hunk 3]
```diff
@@ -76,22 +77,34 @@ export async function generateTypeScriptLibrary(
 
   const shouldGenerateLicense: boolean =
     (await host.GetValue("license-header")) || false;
-
   const hideClients: boolean = (await host.GetValue("hide-clients")) || false;
+  const armLibrary: boolean =
+    (await host.GetValue("azure-arm")) ||
+    (await host.GetValue("openapi-type")) === "arm" ||
+    false;
+  const ignoreNullableOnOptional: boolean =
```
Comment (joheredi) [inline src/typescriptGenerator.ts:85]: I think you can do the following and later on you just need to check for `ignoreNullableOnOptional` without also checking for armLibrary

```typescript
const ignoreNullableOnOptional: boolean = (await host.GetValue("ignore-nullable-on-optional")) ??  (await host.GetValue("azure-arm")) 
```

File: src/generators/modelsGenerator.ts [hunk 4]
```diff
@@ -701,7 +712,14 @@ function getProperties(
       property.name === "siblings"
         ? `${(objectDetails as PolymorphicObjectDetails).unionName}[]`
         : property.type;
-    return property.nullable ? `${typeName} | null` : typeName;
+
+    if (optionsBag.armLibrary || optionsBag.ignoreNullableOnOptional) {
```
Comment (joheredi) [inline src/generators/modelsGenerator.ts:716]: If you do something like I suggested below, you should be able to just check for `optionsBag.ignoreNullableOnOptional`. Which would be safer as you don't need to remember to check `armLibrary` every time you check for `ignoreNullableOnOptional`

**Review by joheredi — APPROVED** [review]
Looks good! Thanks 😄 
Left one suggestion to avoid checking both `ignoreNullableOnOptional` and `armLibrary`

**Pull request merged**